tests/testsuite.py

```diff
@@ -0,0 +1,147 @@
+# pylint: disable=missing-docstring, unused-argument, wrong-import-position, invalid-name, line-too-long
+import html
+import io
+import re
+import sys
+import unittest
+from importlib import import_module
+from types import ModuleType
+from typing import Callable, List, Type
+
+from IPython.core.interactiveshell import InteractiveShell
+from IPython.core.magic import Magics, cell_magic, magics_class
+from IPython.core.magic_arguments import argument, magic_arguments, parse_argstring
+from IPython.display import HTML, display
+
+
+def find_class(namespace: ModuleType, name: str) -> Type:
+    """
+    Recursively find a class in a local namespace by name
+    """
+
+    def find_rec(current: dict, components: List[str]) -> Type:
+        if len(components) == 1:
+            return getattr(current, components[0])
+        return find_rec(getattr(current, components[0]), components[1::])
+
+    return find_rec(namespace, name.split("."))
+
+
+class SubmissionTest(unittest.TestCase):
+    """
+    Base class for submission testing.
+    To implement your problems just subclass this one
+    and add your own test cases.
+    If possible, do not override `__init__` and just add
+    your test methods.
+    """
+
+    def __init__(
+        self,
+        fun: Callable,
```

Comment:
I am not a big fan of short names with little meaning. Maybe we can call it something like "tested_function" or "test_function".